confluent_kafka/src/confluent_kafka.c

```diff
@@ -333,16 +333,20 @@ static PyObject *Message_timestamp (Message *self, PyObject *ignore) {
 }
 
 static PyObject *Message_set_value (Message *self, PyObject *new_val) {
-   Py_DECREF(self->value);
-   self->value = new_val;
-   Py_INCREF(self->value);
+   if (self->value) {
+        Py_DECREF(self->value);
+        self->value = new_val;
```

Comment:
This only sets a new value if there was an old one, which isnt right.
Look at my initial comment.